src/main/java/edu/harvard/iq/dataverse/engine/command/impl/ReturnDatasetToAuthorCommand.java

```diff
@@ -0,0 +1,97 @@
+
+    public ReturnDatasetToAuthorCommand(DataverseRequest aRequest, DvObject anAffectedDvObject, String comment) {
+        super(aRequest, anAffectedDvObject);
+        this.theDataset = (Dataset) anAffectedDvObject;
+        this.comment = comment;
+    }
+
+    @Override
+    public Dataset execute(CommandContext ctxt) throws CommandException {
+        if (theDataset == null) {
+            throw new IllegalCommandException(BundleUtil.getStringFromBundle("dataset.reject.datasetNull"), this);
+        }
+
+        if (!theDataset.getLatestVersion().isInReview()) {
+            throw new IllegalCommandException(BundleUtil.getStringFromBundle("dataset.reject.datasetNotInReview"), this);
+        }
+        /*
+        if(theDataset.getLatestVersion().getReturnReason() == null || theDataset.getLatestVersion().getReturnReason().isEmpty()){
+             throw new IllegalCommandException("You must enter a reason for returning a dataset to the author(s).", this);
+        }
+         */
+        return save(ctxt);
+    }
+
+    public Dataset save(CommandContext ctxt) throws CommandException {
+
+        Timestamp updateTime = new Timestamp(new Date().getTime());
+        theDataset.getEditVersion().setLastUpdateTime(updateTime);
+        // We set "in review" to false because now the ball is back in the author's court.
+        theDataset.getEditVersion().setInReview(false);
+        theDataset.setModificationTime(updateTime);
+        
+        Dataset savedDataset = ctxt.em().merge(theDataset);
+        ctxt.em().flush();
+
+        DatasetVersionUser ddu = ctxt.datasets().getDatasetVersionUser(theDataset.getLatestVersion(), this.getUser());
+        
+        WorkflowComment workflowComment = new WorkflowComment(new WorkflowAction(theDataset.getEditVersion(), RETURN_TO_AUTHOR), comment, (AuthenticatedUser) this.getUser());
+        ctxt.datasets().addWorkflowComment(workflowComment);
+
+        if (ddu != null) {
+            ddu.setLastUpdateDate(updateTime);
+            ctxt.em().merge(ddu);
+        } else {
+            // TODO: This logic to update the DatasetVersionUser was copied from UpdateDatasetCommand and also appears in CreateDatasetCommand, PublishDatasetCommand UpdateDatasetCommand, and SubmitDatasetForReviewCommand. Consider consolidating.
+            DatasetVersionUser datasetDataverseUser = new DatasetVersionUser();
+            datasetDataverseUser.setDatasetVersion(savedDataset.getLatestVersion());
+            datasetDataverseUser.setLastUpdateDate((Timestamp) updateTime);
+            String id = getUser().getIdentifier();
+            id = id.startsWith("@") ? id.substring(1) : id;
+            AuthenticatedUser au = ctxt.authentication().getAuthenticatedUser(id);
+            datasetDataverseUser.setAuthenticatedUser(au);
+            ctxt.em().merge(datasetDataverseUser);
+        }
+        List<AuthenticatedUser> authUsers = ctxt.permissions().getUsersWithPermissionOn(Permission.PublishDataset, savedDataset);
```

Comment:
I'd say still worth fixing. Being inconsistently better is better than being consistently worse.